src/Magnum/Platform/GlfwApplication.cpp

```diff
@@ -178,13 +178,24 @@ void GlfwApplication::staticMouseEvent(GLFWwindow*, int button, int action, int)
     } /* we don't handle GLFW_REPEAT */
 }
 
-void GlfwApplication::staticMouseScrollEvent(GLFWwindow*, double xoffset, double yoffset) {
+void GlfwApplication::staticMouseScrollEvent(GLFWwindow* window, double xoffset, double yoffset) {
     MouseScrollEvent e(Vector2d{xoffset, yoffset});
     _instance->mouseScrollEvent(e);
 
     if(yoffset != 0.0) {
-        MouseEvent e1((yoffset > 0.0) ? MouseEvent::Button::WheelUp : MouseEvent::Button::WheelDown);
-        _instance->mousePressEvent(e);
+        InputEvent::Modifiers mods =
+                (glfwGetKey(window, Int(KeyEvent::Key::LeftShift)) == GLFW_PRESS || glfwGetKey(window, Int(KeyEvent::Key::RightShift)) == GLFW_PRESS)
+                 ? InputEvent::Modifiers{InputEvent::Modifier::Shift} : InputEvent::Modifiers{};
+        if(glfwGetKey(window, Int(KeyEvent::Key::LeftAlt)) == GLFW_PRESS || glfwGetKey(window, Int(KeyEvent::Key::RightAlt)) == GLFW_PRESS) {
+            mods |= InputEvent::Modifier::Alt;
+        }
+        if(glfwGetKey(window, Int(KeyEvent::Key::LeftCtrl)) == GLFW_PRESS || glfwGetKey(window, Int(KeyEvent::Key::RightCtrl)) == GLFW_PRESS) {
+            mods |= InputEvent::Modifier::Ctrl;
+        }
+        if(glfwGetKey(window, Int(KeyEvent::Key::RightSuper)) == GLFW_PRESS) {
+            mods |= InputEvent::Modifier::AltGr;
+        }
+        MouseEvent e1((yoffset > 0.0) ? MouseEvent::Button::WheelUp : MouseEvent::Button::WheelDown, mods);
```

Comment:
I don't see a problem with emulating `WheelUp` and `WheelDown`, that is just a single line in the above example and the same goes for `Sdl2Application`. (At least until we deprecate that and replace it with `ScrollEvent` everywhere. But later. I don't want to change everything now.)

To make myself clear: my concern was not the "emulation", but the crazy handling of scroll event *modifiers* in GLFW. But as there is apparently no other way, we have to live with it.